### PR TITLE
[vuelidate] Fix max call stack size error in Vue 2.7.6-14

### PR DESCRIPTION
**Pull request by IGx89**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/vuejs/vue/issues/12750 and https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/61288
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by IGx89** [comment]
This is an existing failure unrelated to my changes, and in a dependency (Vue) and not this package's types. Appears that Vue's types may use features not present in TypeScript 4.2.

**Comment by IGx89** [comment]
Confirmed, Vue's types only support TypeScript 4.4+: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-4.html#symbol-and-template-string-pattern-index-signatures. How can I tell the tests that...

**Review by janesser — APPROVED** [review]
Looks good to me, one ought think of the test file.

File: types/vuelidate/vuelidate-tests.ts [hunk 1]
```diff
@@ -195,6 +195,16 @@ export class ValidComponent extends Vue {
             return !this.$v.password.minLength
         } else return false
     }
+
+	verifyNoTypeRecursion() {
```
Comment (DangerBotOSS) [inline types/vuelidate/vuelidate-tests.ts:199]: - :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.

**Review by janesser — APPROVED** [review]
nice!

**Comment by IGx89** [comment]
Ready to merge

**Pull request merged**